core/src/main/scala/org/apache/spark/scheduler/TaskSetManager.scala

```diff
@@ -58,14 +58,21 @@ private[spark] class TaskSetManager(
 
   val conf = sched.sc.conf
 
-  /*
-   * Sometimes if an executor is dead or in an otherwise invalid state, the driver
-   * does not realize right away leading to repeated task failures. If enabled,
-   * this temporarily prevents a task from re-launching on an executor where
-   * it just failed.
+  /**
+   * This timeout (specified in milliseconds) is used to prevent tasks from being immediatley
+   * re-launched on an executor where the task has already failed. A task will not be re-launched
+   * on an executor where it has already failed until this amount of time has elapsed since the
+   * failure.  One example of when this is useful is if an executor is in dead and the driver
```

Comment:
extra "in".  (or maybe it was supposed to be "in fact")

I might change to talk about misconfigured executors, that appear to be alive, but can't run tasks.  At least thats the use case I've seen for this.  But no big deal really.